src/audio/kpb.c

```diff
@@ -1101,13 +1112,14 @@ static int kpb_copy(struct comp_dev *dev)
 {
 	int ret = 0;
 	struct comp_data *kpb = comp_get_drvdata(dev);
-	struct comp_buffer *source, *sink;
+	struct comp_buffer *source, *sink = NULL;
```

Comment:
no need to assign NULL here, you have a check `if (!sink)` after every sink assignment below

Reply:
@iganakov it *is* now needed with this version of the code (at least formally) because of the `if` at the new line number 1229 - if that `if` is never entered, `sink` remains unassigned and is checked for `NULL` after the loop. But after the code is changed to take into account the fact, that `kpb->sinks[id]->id == id` always, this initialisation probably won't be needed again.

Reply:
Unfortunately this NULL is necessary, otherwise the PR could not pass all steps due to an error that an uninitialised variable could be used.

Reply:
@fkwasowi best to share the compiler error so that @iganakov and @lyakh can resolve.

Reply:
/home/sof/work/sof.git/src/audio/kpb.c: In function 'kpb_copy':
/home/sof/work/sof.git/src/include/sof/coherent.h:[318](https://github.com/thesofproject/sof/actions/runs/4596130915/jobs/8117209535?pr=7309#step:4:319):10: error: 'sink' may be used uninitialized in this function [-Werror=maybe-uninitialized]
  318 |   c->key = k_spin_lock(&c->lock);
/home/sof/work/sof.git/src/audio/kpb.c:1115:31: note: 'sink' was declared here
 1115 |  struct comp_buffer *source, *sink;
      |                               ^~~~
cc1: all warnings being treated as errors
make[3]: *** [CMakeFiles/sof.dir/build.make:772: CMakeFiles/sof.dir/src/audio/kpb.c.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/Makefile2:1453: CMakeFiles/sof.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:2390: src/arch/xtensa/CMakeFiles/bin.dir/rule] Error 2
make: *** [Makefile:736: bin] Error 2
Error: Process completed with exit code 2.